File: bukkit/elysium-chat-bukkit/src/main/java/com/seventh_root/elysium/chat/bukkit/chatchannel/pipeline/BukkitIRCChatChannelPipelineComponent.java
package com.seventh_root.elysium.chat.bukkit.chatchannel.pipeline;

import com.seventh_root.elysium.api.chat.ChatChannelPipelineComponent;
import com.seventh_root.elysium.api.chat.ChatMessageContext;

import static com.seventh_root.elysium.api.chat.ChatChannelPipelineComponent.Type.POST_PROCESSOR;

public class BukkitIRCChatChannelPipelineComponent extends ChatChannelPipelineComponent {

    private String ircChannel;
    private boolean whitelist;

    public BukkitIRCChatChannelPipelineComponent(String ircChannel, boolean whitelist) {
        this.ircChannel = ircChannel;
        this.whitelist = whitelist;
    }

    @Override
    public Type getType() {
        return POST_PROCESSOR;
    }

    @Override
    public String process(String message, ChatMessageContext context) {
        return message;
    }

    public String getIRCChannel() {
        return ircChannel;
    }

    public void setIRCChannel(String ircChannel) {
        this.ircChannel = ircChannel;
    }

    public boolean isWhitelist() {
        return whitelist;
    }

    public void setWhitelist(boolean whitelist) {
        this.whitelist = whitelist;
    }

}

File: bukkit/elysium-chat-bukkit/src/main/kotlin/com/seventh_root/elysium/chat/bukkit/chatchannel/pipeline/BukkitIRCChatChannelPipelineComponent.kt
package com.seventh_root.elysium.chat.bukkit.chatchannel.pipeline

import com.seventh_root.elysium.chat.bukkit.chatchannel.pipeline.ChatChannelPipelineComponent
import com.seventh_root.elysium.chat.bukkit.chatchannel.pipeline.ChatChannelPipelineComponent.Type.POST_PROCESSOR
import com.seventh_root.elysium.chat.bukkit.context.ChatMessageContext
import com.seventh_root.elysium.chat.bukkit.context.ChatMessagePostProcessContext

class BukkitIRCChatChannelPipelineComponent(var ircChannel: String?, var isWhitelist: Boolean) : ChatChannelPipelineComponent() {

    override val type: ChatChannelPipelineComponent.Type
        get() = POST_PROCESSOR

    override fun process(message: String, context: ChatMessageContext): String {
        return message
    }

    override fun postProcess(message: String, context: ChatMessagePostProcessContext): String {
        return message
    }

}
